src/Pim/Bundle/CatalogBundle/EventSubscriber/LoadProductValuesSubscriber.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\EventSubscriber;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+use Pim\Component\Catalog\Model\ProductInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Load real product values object from the $rawValues field (ie: values in JSON)
+ * when a product is loaded by Doctrine.
+ *
+ * @author    Julien Julien <j.janvier@gmail.com>
+ * @copyright 2017 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * TODO: we could use an Entity Listener instead (need to upgrade bundle to 1.3)
+ * TODO: cf. http://symfony.com/doc/current/bundles/DoctrineBundle/entity-listeners.html
+ * TODO: cf. http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/events.html#entity-listeners
+ */
+class LoadProductValuesSubscriber implements EventSubscriber
+{
+    /** @var ContainerInterface */
+    protected $container;
+
+    /**
+     * @param ContainerInterface $container
+     */
+    public function __construct(ContainerInterface $container)
+    {
+        $this->container = $container;
```

Comment:
here I inject the container to avoid a circular reference
I didn't find any other way to do it
open to every proposal :)

basically we have this each time we try to inject something related to the Doctrine entity manager in a Symfony subscriber